internal/server/handler_v2.go

```diff
@@ -93,62 +91,57 @@ func (s *Server) V2Resolve(
 func (s *Server) V2Node(
 	ctx context.Context, in *pbv2.NodeRequest,
 ) (*pbv2.NodeResponse, error) {
-	arcs, err := v2.ParseProperty(in.GetProperty())
-	if err != nil {
-		return nil, err
-	}
-	if len(arcs) == 1 {
-		arc := arcs[0]
-		direction := util.DirectionOut
-		if !arc.Out {
-			direction = util.DirectionIn
+	if in.GetNextToken() == "" {
+		// When |next_token| in request is empty, there are two cases:
+		// 1. The call does not need pagination.
+		// 2. The call needs pagination, but this is the first call/page.
+		//
+		// To decide which case it is, we call local Mixer, and examine the
+		// |next_token| in the response.
+
+		resp, err := V2NodeCore(ctx, s.store, s.cache, s.metadata, in)
+		if err != nil {
+			return nil, err
 		}
 
-		if arc.SingleProp != "" && arc.Wildcard == "+" {
-			// Examples:
-			//   <-containedInPlace+{typeOf:City}
-			return v2pv.LinkedPropertyValues(
-				ctx,
-				s.store,
-				s.cache,
-				in.GetNodes(),
-				arc.SingleProp,
-				direction,
-				arc.Filter,
-			)
+		// Call remote Mixer and merge only for non-paginated cases,
+		// which are signaled by empty resp.GetNextToken().
+		if resp.GetNextToken() == "" && s.metadata.RemoteMixerURL != "" {
```

Comment:
what if resp.GetNextToken() != ""?

Reply:
I don't quite get the comment from line 107-108, maybe elaborate?

Reply:
Updated with new recursive design.